src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -202,6 +215,15 @@ public ZooKeeperSaslClient getSaslClient() {
 
     private final ZKWatchManager watchManager;
 
+    private ZKClientConfig clientConfig;
+
+    public ZKClientConfig getClientConfig() {
```

Comment:
I'd rather not synchronize it here. It sounds better to check in the constructor whether config is null and instantiate the default object with `new ZKClientConfig()` in the constructor, in which case we can remove this if block from here, or at least keep it here just to assert that `clientConfig`is never null. If we do it, then we can also make `clientConfig`  final.